q/unzip/unzip.q

```diff
@@ -567,6 +575,31 @@
   / get byte count
   c:.finos.unzip.priv.bcount y;
 
+  / check for empty zip
+  / looks for the exact byte vector of a valid zip file containing only
+  /  an end of central directory record with all fields except the
+  /  signature containing only zeroes
+  if[$[c=22;0x504b0506000000000000000000000000000000000000~.finos.unzip.priv.bytes[y;0;c];0b];
```

Comment:
Is this common enough to warrant special handling? Aren't there other possible representations? Wouldn't it be handled as a degenerate case by the existing code?

Reply:
we got one from a vendor, so it definitely needs to be handled, but i can have a look to see if it can be done without such a hard-coded special case